nodemanager/core/RemoteExecutor.cpp

```diff
@@ -530,16 +533,26 @@ void RemoteExecutor::StartHeartbeat(const std::string& callbackUri)
     WriterLock writerLock(&this->lock);
 
     this->nodeInfoReporter =
-        std::unique_ptr<Reporter<json::value>>(
-            new HttpReporter(
+        std::unique_ptr<HttpPeriodicSender>(
+            new HttpPeriodicSender(
                 callbackUri,
                 0,
                 this->NodeInfoReportInterval,
-                [this]() { return this->jobTaskTable.ToJson(); }));
+                methods::Post,
+                [this](http_request& request) {request.set_body(return this->jobTaskTable.ToJson()); return true;},
+                nullptr));
 
     this->nodeInfoReporter->Start();
 }
 
+void RemoteExecutor::StartHostsManager(const std::string& callbackUri)
+{
+    WriterLock writerLock(&this->lock);
+
+    this->hostsManager = std::unique_ptr<HostsManager>(callbackUri);
```

Comment:
this->hostsManager = std::unique_ptr<HostsManager>(new HostsManager(callbackUri));